barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -3528,8 +3528,15 @@ std::vector<Row> AvmTraceBuilder::finalize()
     for (size_t i = 0; i < main_trace_size; i++) {
         main_trace[i].main_sel_execution_row = FF(1);
     }
-    // This selector corresponds to the last row of the EXECUTION trace.
-    main_trace.back().main_sel_last = FF(1);
+
+    // Toggle start of the execution row
+    if (main_trace_size > 0) {
```

Comment:
I think you always need to set this. If the trace is not big enough, maybe you need to push a row. The reason is that this will be in the VK, is a constant column, so it has to always be the same in the extended 2^20+ trace.

Reply:
Good catch. I moved this to after `main_trace.emplace_back(end_exec_row);` and therefore we are sure that the trace contains at least one entry.